frontend/src/components/playground/AceEditorComponent.tsx

```diff
@@ -0,0 +1,256 @@
+"use client";
+
+import { useState, useEffect, useRef, useCallback } from "react";
+import AceEditor, { IAnnotation } from "react-ace";
+import { usePlayground } from "@/context/PlaygroundContext";
+import { usePlaygroundActions } from "@/hooks/usePlaygroundActions";
+import { SaveState, SaveStatus } from "@/types/SaveStatus";
+
+// Import ace-builds core
+import ace from "ace-builds";
+import beautify from "ace-builds/src-noconflict/ext-beautify";
+
+// Import modes
+import "ace-builds/src-noconflict/mode-json";
+import "ace-builds/src-noconflict/mode-javascript";
+import "ace-builds/src-noconflict/theme-github";
+import "ace-builds/src-noconflict/ext-language_tools";
+// This seems like it should be imported from the line above, but there are missing values (like showing highlighted text) without this
+import "ace-builds/css/theme/github.css";
+
+// Import workers
+import jsonWorkerUrl from "ace-builds/src-noconflict/worker-json";
+import javascriptWorkerUrl from "ace-builds/src-noconflict/worker-javascript";
+
+// Configure Ace to use the imported workers
+ace.config.setModuleUrl("ace/mode/json_worker", jsonWorkerUrl);
+ace.config.setModuleUrl("ace/mode/javascript_worker", javascriptWorkerUrl);
+
+interface AceEditorComponentProps {
```

Comment:
[Design] I've got a preference towards requiring all parameters, could we removal the `?` from many of these members?  This way we have fewer edges cases to handle within the component